OpenInFirefoxClient/OpenInFirefoxControllerObjC.m

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+#import <UIKit/UIKit.h>
+#import "OpenInFirefoxControllerObjC.h"
+
+ static NSString *const firefoxScheme = @"firefox:";
+ static NSString *const myAppScheme = @"firefoxopeninfirefoxclient://";
+
+@implementation OpenInFirefoxControllerObjC
+
+//creates shared instance of the controller
+ + (OpenInFirefoxControllerObjC *)sharedInstance {
+  static OpenInFirefoxControllerObjC *sharedInstance;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    sharedInstance = [[self alloc] init];
+  });
+  return sharedInstance;
+}
+
+//custom function that does complete percent escape for constructing the query URL
```

Comment:
Nit: For consistency, and to make comments easier to read, please use full sentences with capitalization and punctuation throughout.